### PR TITLE
Fix `mypy` warnings for `astroid/builder` and `astroid/nodes/as_string`

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

File: astroid/builder.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def delayed_assattr(self, node):
 
 
 def build_namespace_package_module(name, path: str) -> nodes.Module:
-    return nodes.Module(name, doc="", path=path, package=True)
+    return nodes.Module(name, doc="", path=[path], package=True)
```
Comment (cdce8p) [inline astroid/builder.py:267]: This is probably a true bug fix. If `path` was set as `str`, `path[0]` would access just the first char and not the complete string. Can you create a separate PR for it, ideally with changelog entry. We should also test how this effects existing code.

**Review by cdce8p — CHANGES_REQUESTED** [review]

File: astroid/builder.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply astroid/builder.py:267]: If this is a bug fix maybe we can put it in 2.8.6.

Comment (DanielNoord) [reply astroid/builder.py:267]: I looked into it. I don't think `path: str` is correct here. Looking at how this is called the `path` always comes from `ModuleSpec.submodule_search_locations` which is always a `List[str]`. 

You can run the test `test_nested_namespace_import` to debug this.

Thus, this change is actually incorrect and the typing should be updated. I will push a new commit.

Comment (cdce8p) [reply astroid/builder.py]: That does make sense! I guess otherwise namespace packages would be broken completely.

**Review by cdce8p — APPROVED** [review]

**Pull request merged**